components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/registry/RegistryManager.java

```
@@ -348,6 +349,28 @@ public void persistServiceGroup(ServiceGroup servicegroup) {
         }
     }
 
+
+    public void updateServiceGroup(ServiceGroup serviceGroup) throws InvalidServiceGroupException {
+        try {
+            if (serviceGroup == null || StringUtils.isEmpty(serviceGroup.getName())) {
+                throw new IllegalArgumentException("Cartridge group or group name cannot be null");
+            }
+            if (getServiceGroup(serviceGroup.getName()) == null) {
+                throw new InvalidServiceGroupException(String.format("Cartridge group does not exist: " +
+                        "[cartridge-group] %s", serviceGroup.getName()));
+            }
+
+            persistServiceGroup(serviceGroup);
+
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Updated cartridge group: [group-name] %s", serviceGroup.getName()));
+            }
+        } catch (Exception e) {
```

Yes please throw an exception with a proper message.